iotfunctions/pipeline.py

```diff
@@ -2357,15 +2372,16 @@ def apply_scope(self, df, stage):
                     eval_expression += 'df[' + dimension_name + '].isin(' + dimension_value + ')' + '&' if dimension_count != 0 else ''
```

Comment:
I think there should be OR (|) operation instead of AND (&) The dimension value can not be both at the same time.

Reply:
@herleraja 
I thought this is how we'll handle multiple dimensions
`df['manufacturer'].isin(['Rentech']) & df['firmware'].isin(['2.1'])`

Reply:
what happens if you have only one value for particular entity or device id ? 

example: 

device id  | manufacture | firmware
device1     |    Rentech     | ---------
device2     |    ----------  | ---------
device3     |    Rentech     | 2.1
device4     |    --------     | 2.1

as per the above logic, it will select only one record but it should select 3 records.  So only I was insisting on OR (|) instead of AND (&)

Reply:
@herleraja As per today's playback feedback, I need to check with @rsivaibm about **AND**  or **OR**